tokio-util/src/task/join_map.rs

```diff
@@ -716,7 +698,9 @@ where
     #[inline]
     pub fn shrink_to_fit(&mut self) {
         self.hashes_by_task.shrink_to_fit();
-        self.tasks_by_key.shrink_to_fit();
+        let hash_builder = self.hashes_by_task.hasher();
+        self.tasks_by_key
+            .shrink_to_fit(|(k, _)| hash_one(hash_builder, k));
```

Comment:
I'm confused. Why is `hashes_by_task` needed inside `tasks_by_key` here?

Reply:
We only need the hasher inside `hashes_by_task`. It saves us from having to store a separate instance of the hasher.

An alternative solution could be to have a separate `hasher: S`, and then make the `HashMap<Id, u64>` use a cheaper hash function like a simple fibonacci hash (so no state is needed). That might be overkill though.